mm2src/coins/nft/storage/wasm/wasm_storage.rs

```diff
@@ -1,132 +1,101 @@
-#[derive(Clone)]
-pub struct IndexedDbNftStorage {
-    /// The underlying shared database instance for caching NFT data.
-    db: SharedDb<NftCacheIDB>,
+fn take_nft_according_to_paging_opts(
+    mut nfts: Vec<Nft>,
+    max: bool,
+    limit: usize,
+    page_number: Option<NonZeroUsize>,
+) -> WasmNftCacheResult<NftList> {
+    let total_count = nfts.len();
+    nfts.sort_by(|a, b| b.block_number.cmp(&a.block_number));
+    let (offset, limit) = get_offset_limit(max, limit, page_number, total_count);
+    Ok(NftList {
+        nfts: nfts.into_iter().skip(offset).take(limit).collect(),
+        skipped: offset,
+        total: total_count,
+    })
 }
 
-impl IndexedDbNftStorage {
-    /// Construct a new `IndexedDbNftStorage` using the given MM context.
-    ///
-    /// This method ensures that a proper NFT context (`NftCtx`) exists within the MM context
-    /// and initializes the underlying storage as required.
-    pub fn new(ctx: &MmArc) -> MmResult<Self, CreateNftStorageError> {
-        let nft_ctx = NftCtx::from_ctx(ctx).map_to_mm(CreateNftStorageError::Internal)?;
-        Ok(IndexedDbNftStorage {
-            db: nft_ctx.nft_cache_db.clone(),
-        })
-    }
-
-    /// Lock the underlying database to ensure exclusive access, maintaining data consistency during operations.
-    async fn lock_db(&self) -> WasmNftCacheResult<NftCacheIDBLocked<'_>> {
-        self.db.get_or_initialize().await.mm_err(WasmNftCacheError::from)
-    }
-
-    fn take_nft_according_to_paging_opts(
-        mut nfts: Vec<Nft>,
-        max: bool,
-        limit: usize,
-        page_number: Option<NonZeroUsize>,
-    ) -> WasmNftCacheResult<NftList> {
-        let total_count = nfts.len();
-        nfts.sort_by(|a, b| b.block_number.cmp(&a.block_number));
-        let (offset, limit) = get_offset_limit(max, limit, page_number, total_count);
-        Ok(NftList {
-            nfts: nfts.into_iter().skip(offset).take(limit).collect(),
-            skipped: offset,
-            total: total_count,
-        })
-    }
-
-    fn filter_nfts<I>(nfts: I, filters: Option<NftListFilters>) -> WasmNftCacheResult<Vec<Nft>>
-    where
-        I: Iterator<Item = NftListTable>,
-    {
-        let mut filtered_nfts = Vec::new();
-        for nft_table in nfts {
-            let nft = nft_details_from_item(nft_table)?;
-            if let Some(filters) = &filters {
-                if filters.passes_spam_filter(&nft) && filters.passes_phishing_filter(&nft) {
-                    filtered_nfts.push(nft);
-                }
-            } else {
+fn filter_nfts<I>(nfts: I, filters: Option<NftListFilters>) -> WasmNftCacheResult<Vec<Nft>>
+where
+    I: Iterator<Item = NftListTable>,
+{
+    let mut filtered_nfts = Vec::new();
+    for nft_table in nfts {
+        let nft = nft_details_from_item(nft_table)?;
+        if let Some(filters) = &filters {
+            if filters.passes_spam_filter(&nft) && filters.passes_phishing_filter(&nft) {
```

Comment:
Please consider using `match` any time when you need to handle more than one enum variant.